### PR TITLE
Created CHIMERA and related simulation data

### PR DESCRIPTION
**Pull request by edefilippo**
Created CHIMERA and related simulation data, first version compatible with R3BRoot/dev

### CODE REVIEW

File: asydata/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Create a library called "libR3BasyData" which includes the source files given in
```
Comment (jose-luis-rs) [inline asydata/CMakeLists.txt:1]: the name of this lib should be "libR3B**A**syData"

File: asydata/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+chimeraData/R3BAsyChimeraHitData.cxx
+)
+
+
+# fill list of header files from list of source files
+# by exchanging the file extension
+CHANGE_FILE_EXTENSION(*.cxx *.h HEADERS "${SRCS}")
+SET(HEADERS ${HEADERS} ${R3BROOT_SOURCE_DIR}/r3bdata/R3BDetectorList.h )
+
+set(LINKDEF  R3BasyDataLinkDef.h)
```
Comment (jose-luis-rs) [inline asydata/CMakeLists.txt:33]: R3B**A**syDataLinkDef.h

File: asydata/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+set(LIBRARY_NAME R3BasyData)
```
Comment (jose-luis-rs) [inline asydata/CMakeLists.txt:34]: R3B**A**syData

File: asydata/R3BasyDataLinkDef.h [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// 2017/04/07 12:02:26 hapol
```
Comment (jose-luis-rs) [inline asydata/R3BasyDataLinkDef.h:1]: this file is excluded from the clang-format, for this you must add this line:
// clang-format off

File: asydata/R3BasyDataLinkDef.h [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+#ifdef __CINT__
+
+#pragma link off all globals;
+#pragma link off all classes;
+#pragma link off all functions;
+
+// Point classes for simulation
+#pragma link C++ class R3BAsyChimeraPoint + ;
```
Comment (jose-luis-rs) [inline asydata/R3BasyDataLinkDef.h:10]: without space:
#pragma link C++ class R3BAsyChimeraPoin**t+;**

File: asydata/chimeraData/R3BAsyChimeraHitData.h [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+     *@param fY         Position Y in [mm]
+     *@param fTof      Tof [ns]
+     **/
+    R3BAsyChimeraHitData(Int_t DetId, Double_t ELoss, Double_t Fast, Double_t Slow, Double_t ToF);
+
+    /** Destructor **/
+    virtual ~R3BAsyChimeraHitData() {}
+
+    /** Accessors **/
+    inline const Int_t& GetDetId() const { return fDetId; }
```
Comment (jose-luis-rs) [inline asydata/chimeraData/R3BAsyChimeraHitData.h:24]: With this line is enough:
Int_t GetDetId() const { return fDetId; }
remove the **inline const**

Comment (jose-luis-rs) [reply asydata/chimeraData/R3BAsyChimeraHitData.h:24]: in general

File: asydata/chimeraData/R3BAsyChimeraPoint.cxx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    fSlow= slow;
+}
+// -------------------------------------------------------------------------
+
+// -----   Destructor   ----------------------------------------------------
+R3BAsyChimeraPoint::~R3BAsyChimeraPoint() {}
+// -------------------------------------------------------------------------
+
+// -----   Public method Print   -------------------------------------------
+void R3BAsyChimeraPoint::Print(const Option_t* opt) const
```
Comment (jose-luis-rs) [inline asydata/chimeraData/R3BAsyChimeraPoint.cxx:59]: The print function is not used anymore, please, remove it. See for instance:
https://github.com/R3BRootGroup/R3BRoot/blob/dev/r3bdata/alpideData/R3BAlpidePoint.cxx
https://github.com/R3BRootGroup/R3BRoot/blob/dev/r3bdata/alpideData/R3BAlpidePoint.h

File: chimera/AsyChimLinkDef.h [hunk 8]
```diff
@@ -0,0 +1,22 @@
+#ifdef __CINT__
```
Comment (jose-luis-rs) [inline chimera/AsyChimLinkDef.h:1]: Please, add:
// clang-format off

File: chimera/AsyChimLinkDef.h [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+#pragma link off all globals;
+#pragma link off all classes;
+#pragma link off all functions;
+
+#pragma link C++ class R3BAsyChimera + ;
```
Comment (jose-luis-rs) [inline chimera/AsyChimLinkDef.h:7]: without spaces: ...Chimera + ; ---> ...Chimera+;

File: chimera/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,41 @@
+# Create a library called "libasySofChim" which includes the source files given in
```
Comment (jose-luis-rs) [inline chimera/CMakeLists.txt:1]: should be "libR3BAsyChim"

File: chimera/R3BAsyChimera.cxx [hunk 11]
```diff
@@ -0,0 +1,369 @@
...
+                     fLength,
+                     fELoss,
+		             fLightYield,
+		             fSlow);
+
+            // Increment number of AsyChimeraPoints for this track
+            R3BStack* stack = (R3BStack*)gMC->GetStack();
+            stack->AddPoint(kCHIMERA);
+
+            // Cal Data for each plastic
```
Comment (jose-luis-rs) [inline chimera/R3BAsyChimera.cxx:217]: Please, remove the commented code!

Comment (jose-luis-rs) [reply chimera/R3BAsyChimera.cxx:217]: Please, update this class according to:
https://github.com/R3BRootGroup/sofia/blob/dev/tofwall/sim/R3BSofTofW.cxx

File: chimera/R3BAsyChimeraContFact.cxx [hunk 12]
```diff
@@ -0,0 +1,83 @@
+// ----------------------------------------------------------------------
+//								    -----
+//  Factory for the parameter containers in libR3BAsyChimera           -----
+//         Created 03/11/19  by J.L. Rodriguez-Sanchez              -----
+//								    -----
+// ----------------------------------------------------------------------
+
```
Comment (jose-luis-rs) [inline chimera/R3BAsyChimeraContFact.cxx:7]: Please, update this class according to:
https://github.com/R3BRootGroup/R3BRoot/blob/dev/twim/pars/R3BTwimContFact.cxx
https://github.com/R3BRootGroup/R3BRoot/blob/dev/twim/pars/R3BTwimContFact.h

**Review by jose-luis-rs — CHANGES_REQUESTED** [review]
Please, don't forget to add the new folders
-asydata
-chimera
to the main CMakeLists.txt file:
https://github.com/R3BRootGroup/asyeos/blob/dev/CMakeLists.txt

**Comment by edefilippo** [comment]
Closed after doing all requested changes. We open a new one just after

**Pull request closed without merging**